src/codegen/codegen-expression.lisp

```diff
@@ -224,8 +224,9 @@
             ,(codegen-expression (node-bind-body expr) current-function env)))))))
 
 (defun find-constructor (initform env)
-  (and (node-direct-application-p initform)
-       (tc:lookup-constructor env (node-direct-application-rator initform) :no-error t)))
+  (if (or (node-application-p initform) (node-direct-application-p initform))
+      (and (node-rator-name initform)
+           (tc:lookup-constructor env (node-rator-name initform) :no-error t))))
```

Comment:
NIL as else branch? This code looks like it might be a mistake, but probably isn't.